src/lib/utils/mem_ops.h

```diff
@@ -162,6 +162,20 @@ inline constexpr To typecast_copy(const FromT* src) noexcept
    return dst;
 }
 
+template <class To>
+inline constexpr To typecast_copy(std::span<const uint8_t, sizeof(To)> src) noexcept
+   requires std::is_trivial_v<To>
+{
+   return typecast_copy<To>(src.data());
+}
+
+template <class FromT>
+inline constexpr void typecast_copy(std::span<uint8_t, sizeof(FromT)> out, FromT in)
+   requires std::is_trivial_v<typename std::decay_t<FromT>>
+{
+   typecast_copy(out.data(), in);
+}
+
```

Comment:
In fact, maybe we should turn this around. Put the implementation w/ `memcpy` in here and let the existing bare-pointer overloads call those. I did it like that for the endian-conversions.

Reply:
done.